Mathlib/Topology/Compactness/Compact.lean

```diff
@@ -501,6 +505,28 @@ theorem exists_subset_nhds_of_isCompact' [Nonempty ι] {V : ι → Set X}
   have : ¬⋂ i : ι, V i ⊆ W := by simpa [← iInter_inter, inter_compl_nonempty_iff]
   contradiction
 
+omit [TopologicalSpace X] in
+/--
+**Alexander's subbasis theorem**. Suppose `X` is a topological space with a subbasis `S` and `s` is
+a subset of `X`. Then `s` is compact if for any open cover of `s` with all elements taken from `S`,
+there is a finite subcover.
+-/
+theorem isCompact_generateFrom {S : Set (Set X)} (s : Set X)
+    (h : ∀ P ⊆ S, s ⊆ ⋃₀ P → ∃ Q ⊆ P, Q.Finite ∧ s ⊆ ⋃₀ Q) :
+    @IsCompact X (generateFrom S) s := by
```

Comment:
I think this should be a theorem about arbitrary topologies with a subbasis, instead of only topologies that are defeq to `generateFrom S`.

Reply:
Since there are [many theorems](https://loogle.lean-lang.org/?q=%7C-+_+%3D+TopologicalSpace.generateFrom+_) in mathlib with `_ = generateFrom _` as conclusion, this is probably a good idea. There are also [6 theorems](https://loogle.lean-lang.org/?q=_+%3D+TopologicalSpace.generateFrom+_+%E2%86%92+_) that uses it as an assumption.

Reply:
Should there be a predicate `IsSubbasis s := ‹_› = generateFrom s`?

Reply:
Sounds good to me. But as a follow-up PR. This one can just take the raw assumption and add a TODO to use `IsSubbasis` later.